yara-x/src/modules/macho/mod.rs

```diff
@@ -0,0 +1,2349 @@
+    let fat_arch_size = if fat_is_32(header.magic) {
+        std::mem::size_of::<FatArch32>()
+    } else {
+        std::mem::size_of::<FatArch64>()
+    };
+
+    // Loop through the nested array of fat_archs
+    for i in 0..header.nfat_arch as usize {
+        let arch_data = &remaining_data[i * fat_arch_size..];
+
+        // Parse 32bit FAT headers
+        if fat_is_32(header.magic) {
+            let (_, arch) = parse_fat_arch_32(arch_data)
+                .map_err(|e| MachoError::ParsingError(format!("{:?}", e)))?;
+
+            let fat_arch_entry = FatArch {
+                cputype: Some(arch.cputype),
+                cpusubtype: Some(arch.cpusubtype),
+                offset: Some(arch.offset as u64),
+                size: Some(arch.size as u64),
+                align: Some(arch.align),
+                ..Default::default()
+            };
+            macho_proto.fat_arch.push(fat_arch_entry);
+
+            // Return Overflow error in case offset index or end index is bigger then data size
+            let end_index =
+                arch.offset
+                    .checked_add(arch.size)
+                    .ok_or(MachoError::Overflow)? as usize;
+
+            if arch.offset as usize > data.len() || end_index > data.len() {
+                return Err(MachoError::Overflow);
+            }
+
+            // Parse nested data as basic Mach-O file
+            let nested_file =
+                parse_macho_file(&data[arch.offset as usize..end_index])?;
+            macho_proto.file.push(nested_file);
+        // Parse 64bit FAT headers
+        } else {
+            let (_, arch) = parse_fat_arch_64(arch_data)
+                .map_err(|e| MachoError::ParsingError(format!("{:?}", e)))?;
+
+            let fat_arch_entry = FatArch {
+                cputype: Some(arch.cputype),
+                cpusubtype: Some(arch.cpusubtype),
+                offset: Some(arch.offset),
+                size: Some(arch.size),
+                align: Some(arch.align),
+                reserved: Some(arch.reserved),
+                ..Default::default()
+            };
+            macho_proto.fat_arch.push(fat_arch_entry);
+
+            // Return Overflow error in case offset index or end index is bigger then data size
+            let end_index =
+                arch.offset
+                    .checked_add(arch.size)
+                    .ok_or(MachoError::Overflow)? as usize;
+
+            if arch.offset as usize > data.len() || end_index > data.len() {
+                return Err(MachoError::Overflow);
+            }
+
+            // Parse nested data as basic Mach-O file
+            let nested_file =
+                parse_macho_file(&data[arch.offset as usize..end_index])?;
+            macho_proto.file.push(nested_file);
+        }
+    }
+
+    Ok(())
+}
+
+/// Public wrapper for Mach-O FAT files fuzz testing
+#[cfg(fuzzing)]
+pub fn pub_parse_fat_macho_file(
```

Comment:
Instead of having this wrapper I would simply make `parse_fat_macho_file` public. At the end of the day the issue is that we are forced to publish an internal function, no matter if its only a wrapper or the real function. 

One important detail is marking the function with `#[doc(hidden)]` so that it doesn't appear in public documentation.

Reply:
Fixed, thank you